### PR TITLE
[socket] Add RBS file for "socket"

### PR DESCRIPTION
**Pull request by HoneyryderChuck**
Added RBS signatures for the "socket" package (no tests).

### CODE REVIEW

File: core/socket.rbs [hunk 1]
```diff
@@ -0,0 +1,4348 @@
...
+  # *   `socketoption` is an instance of Socket::Option
+  #
+  #
+  # ### Examples
+  #
+  # Some socket options are integers with boolean values, in this case #setsockopt
+  # could be called like this:
+  #     sock.setsockopt(:SOCKET, :REUSEADDR, true)
+  #     sock.setsockopt(Socket::SOL_SOCKET,Socket::SO_REUSEADDR, true)
+  #     sock.setsockopt(Socket::Option.bool(:INET, :SOCKET, :REUSEADDR, true))
```
Comment (soutaro) [inline core/socket.rbs:441]: > __[Goodcheck]__ Prefer `boolish` over `bool` for method arguments and block return values ([view](https://sider.review/gh/repos/174796313/pulls/699#issue-5728785))
>
> <details>
>
> See the doc below:
> https://github.com/ruby/rbs/blob/78d04a2db0f1c4925d2b13c2939868edf9465d6c/docs/syntax.md#bool-or-boolish
>
> | Rule |
> | ---- |
> | `rbs.prefer_boolish` |
>
> You can [close this issue](https://sider.review/gh/repos/174796313/pulls/699#issue-5728785) if no need to fix it. [Learn more](https://help.sider.review/getting-started/working-with-issues).
>
> </details>

This is a false detection. Closing.

File: core/socket.rbs [hunk 2]
```diff
@@ -0,0 +1,4348 @@
...
+  #         p s1.read          #=> "pong\n"
+  #
+  #     }
+  #
+  #
+  def shutdown: (Symbol | String | Integer flags) -> void
+
+  private
+
+  def __recv_nonblock: (untyped, untyped, untyped, untyped) -> untyped
```
Comment (soutaro) [inline core/socket.rbs:494]: 👍

**Review by soutaro — CHANGES_REQUESTED** [review]
I really appreciate this great work! 🙏 

I want to ask you to change the file name.

* Could you move the file under `stdlib/socket/0` directory, because `socket` is not built-in?
* How about splitting files for each classes, like `basic_socket.rbs`, `socket.rbs`, ...? (This is optional but would make reading and future updates easier.)

File: core/socket.rbs (hunk 2 above)
Comment (HoneyryderChuck) [reply core/socket.rbs]: fine to leave it as is, I suppose? :)

**Comment by HoneyryderChuck** [comment]
@soutaro thx. Just committed the changes you proposed. hope it's fine.

File: core/socket.rbs (hunk 2 above)
Comment (soutaro) [reply core/socket.rbs]: Yes. It's fine to leave the defs here.

**Review by soutaro — APPROVED** [review]
Thank you so much! 👍

File: stdlib/socket/0/socket.rbs [hunk 3]
```diff
@@ -0,0 +1,2687 @@
...
+  #
+  # *protocol* is optional and should be a protocol defined in the domain. If
+  # protocol is not given, 0 is used internally.
+  #
+  #     Socket.new(:INET, :STREAM) # TCP socket
+  #     Socket.new(:INET, :DGRAM)  # UDP socket
+  #     Socket.new(:UNIX, :STREAM) # UNIX stream socket
+  #     Socket.new(:UNIX, :DGRAM)  # UNIX datagram socket
+  #
+  def initialize: (Symbol domain, Symbol socktype, ?Integer protocol) -> untyped
```
Comment (tisonkun) [inline stdlib/socket/0/socket.rbs:1266]: It seems we can write `socket = Socket.new(Socket::AF_INET, Socket::SOCK_STREAM, 0)` where the parameters are `Integer`?

**Pull request merged**